tests/test_deploy_github_pages.py

```diff
@@ -0,0 +1,277 @@
+import pytest
+from subprocess import run
+from tempfile import TemporaryDirectory
+import os
+import exasol_sphinx_github_pages_generator.deploy_github_pages as deploy_github_pages
+
+# TODO change to oauth2?
+# todo write readme
+# todo change source_branch to main
+
+
+def remove_branch(branch_name):
+    target_branch_exists = run(["git", "show-branch", f"remotes/origin/{branch_name}"], capture_output=True,
+                               text=True)
+    # remove remote target branch if exists:
+    if target_branch_exists.returncode == 0:
+        run(["git", "push", "-d", "origin", branch_name], check=True)
+
+
+def setup_test_repo():
+    user_access_token = os.environ.get("MAuserPAT")
+    user_name = os.environ.get("MAuserName")
+    run(["git", "clone", f"https://{user_access_token}@github.com/exasol/sphinx-github-pages-generator-test.git"],
+        check=True)
+    os.chdir("sphinx-github-pages-generator-test")
+    run(["git", "remote", "set-url", "origin",
+         f"https://{user_name}:{user_access_token}@github.com/exasol/sphinx-github-pages-generator-test.git"],
+        check=True)
+
+    run(["git", "config", "--local", "user.email", f"{user_name}@exasol.com"], check=True)
+    run(["git", "config", "--local", "user.name", user_name], check=True)
+    return user_name, user_access_token
+
+
+def test_remote_branch_creation():
+    with TemporaryDirectory() as tempdir:
+        os.chdir(tempdir)
+        setup_test_repo()
+        doc_dir = run(["git", "rev-parse", "--show-toplevel"], capture_output=True, text=True, check=True)
+        os.chdir(f"{doc_dir.stdout[:-1]}/doc")
```

Comment:
Maybe add some comment here wha'ts going on. I believe the test repo already has some content which is being used here.

Reply:
done